msdm/core/distributions/dictdistribution.py

```diff
@@ -4,7 +4,11 @@
 
 
 class UniformDistribution(FiniteDistribution):
-    def __init__(self, support):
+    def __init__(self, support, check_unique=True):
+        if check_unique:
+            assert len(support) == len(set(support)), (
```

Comment:
I avoid adding `set(support)` as an attribute b/c `sample()` is implemented as `rng.choice(self._support)`, which requires that `self._support` be indexable.